File: kcp-impl/compiler-kcp/src/main/java/com/bennyhuo/kotlin/kcp/deepcopy/compiler/ErrorsDeepCopy.java
package com.bennyhuo.kotlin.kcp.deepcopy.compiler;

import com.bennyhuo.kotlin.deepcopy.compiler.kcp.checker.DefaultErrorMessagesDeepCopy;
import com.intellij.psi.PsiElement;

import org.jetbrains.kotlin.diagnostics.DiagnosticFactory1;
import org.jetbrains.kotlin.diagnostics.Errors;

import static org.jetbrains.kotlin.diagnostics.Severity.WARNING;

public interface ErrorsDeepCopy {
    DiagnosticFactory1<PsiElement, String> ELEMENT_NOT_IMPLEMENT_DEEPCopyable = DiagnosticFactory1.create(WARNING);

    @SuppressWarnings("UnusedDeclaration")
    Object _initializer = new Object() {
        {
            Errors.Initializer.initializeFactoryNamesAndDefaultErrorMessages(ErrorsDeepCopy.class, DefaultErrorMessagesDeepCopy.INSTANCE);
        }
    };

}

File: kcp-impl/compiler-kcp/src/main/kotlin/com/bennyhuo/kotlin/deepcopy/compiler/kcp/checker/ErrorsDeepCopy.kt
package com.bennyhuo.kotlin.deepcopy.compiler.kcp.checker

import com.intellij.psi.PsiElement
import org.jetbrains.kotlin.diagnostics.DiagnosticFactory1
import org.jetbrains.kotlin.diagnostics.Errors
import org.jetbrains.kotlin.diagnostics.Severity

interface ErrorsDeepCopy {
    companion object {
        @JvmField
        val ELEMENT_NOT_IMPLEMENT_DEEPCOPYABLE = DiagnosticFactory1.create<PsiElement, String>(Severity.WARNING)

        init {
            Errors.Initializer.initializeFactoryNamesAndDefaultErrorMessages(
                ErrorsDeepCopy::class.java,
                DefaultErrorMessagesDeepCopy
            )
        }
    }
}